pkg/retry/retry.go

```diff
@@ -186,22 +189,19 @@ func (s *Server) handleRetry() http.Handler {
 
 			// every event from newCheckpoint to lastCheckpoint has been processed,
 			// overwrite the checkpoint
-			if processed {
```

Comment:
this variable is not necessary, if we get this far we should update the checkpoint